analyzer.go

```diff
@@ -55,6 +61,25 @@ func (a *Analyzer) Analyze() (platform.AnalyzedMetadata, error) {
 	}, nil
 }
 
+func retrieveCacheMetadata(cache Cache, logger Logger) (platform.CacheMetadata, error) {
```

Comment:
I don't think this is the best place for this function.
What do you think about moving it to a common file (since both the analyzer and the restorer are using it)?

Reply:
I'll move it into layer_metadata_restorer.go since both phases are using that struct. It feels too specific for utils.go